docs/development/development-environment-docker/README.md

```diff
@@ -12,114 +12,159 @@ And nothing else!
 
 To get right into it and just start the application you can just do the following:
 
-```
+```bash
 git clone https://github.com/opf/openproject.git
 cd openproject
-bin/compose setup
-bin/compose start
+cp .env.example .env
+cp docker-compose.override.example.yml docker-compose.override.yml
+docker compose run --rm backend setup
+docker compose run --rm frontend npm install
+docker compose up -d frontend
 ```
 
 Once the containers are done booting you can access the application under `http://localhost:3000`.
 
 ### Tests
 
-You can run tests using `bin/compose rspec`. You can run specific tests too. For instance:
+You can run tests inside the `backend-test` container. You can run specific tests, too.
 
-```
-bin/compose rspec spec/features/work_package_show_spec.rb
+```bash
+# Run all tests (not recommended)
+docker compose run --rm backend-test bundle exec rspec
+
+# Run the specified test
+docker compose run --rm backend-test bundle exec rspec spec/features/work_package_show_spec.rb
 ```
 
 ***
 
 More details and options follow in the next section.
 
 
-> **Note**: docker-compose needs access to at least 4GB of RAM. E.g. for Mac, this requires to [increase the default limit of the virtualized host.](https://docs.docker.com/docker-for-mac/) Signs of lacking memory include an "Exit status 137" in the frontend container.
+> **Note**: docker compose needs access to at least 4GB of RAM. E.g. for Mac, this requires
+> to [increase the default limit of the virtualized host](https://docs.docker.com/docker-for-mac/).
+> Signs of lacking memory include an "Exit status 137" in the frontend container.
 
 ## Step-by-step Setup
 
 ### 1) Checkout the code
 
-First you will need to checkout the code as usual.
+First you will need to check out the code as usual.
 
-```
+```bash
 git clone https://github.com/opf/openproject.git
 ```
 
-This will checkout the dev branch in `openproject`. **Change into that directory.**
+This will check out the dev branch in `openproject`. **Change into that directory.**
 
 If you have OpenProject checked out already make sure that you do not have a `config/database.yml`
-as that will interfere with the database connection inside of the docker containers.
+as that will interfere with the database connection inside the docker containers.
 
 ### 2) Configure environment
 
 Copy the env example to `.env`
 
-```
+```bash
 cp .env.example .env
 ```
 
-Afterwards, set the environment variables to your liking. `DEV_UID` and `DEV_GID` are required to be set so your project
+Afterward, set the environment variables to your liking. `DEV_UID` and `DEV_GID` are required to be set so your project
 directory will not end up with files owned by root.
 
-Both `docker-compose` and `bin/compose` will load the env from this file.
+`docker compose` will load the env from this file.
 
-### 3) Setup database and install dependencies
+You also will want to create a `docker-compose.override.yml` file, which can contain the port exposure of you
```

Comment:
Perhaps you meant "for your containers" here?

Reply:
changed to `for your`